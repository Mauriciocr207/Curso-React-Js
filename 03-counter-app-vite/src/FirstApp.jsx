import { string } from "prop-types";

export default function FirstApp({
  title = "No hay titulo",
  sub = "No hay subtitulo"
}) {
  if(!title) {
    console.error("Not title");
  }
  return (<>
    <h1 data-testid="FirstApp.title">{title}</h1>
    <p data-testid="FirstApp.subtitle">{sub}</p>
    <p data-testid="FirstApp.subtitle">{sub}</p>
  </>)
}

FirstApp.propTypes = {
  title: string.isRequired,
  sub: string.isRequired,
}