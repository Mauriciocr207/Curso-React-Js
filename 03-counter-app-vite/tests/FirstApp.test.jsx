import { render, screen } from "@testing-library/react"
import '@testing-library/jest-dom'
import FirstApp from "../src/FirstApp"

describe('FirstApp test', () => {

    const title = "Hola, Soy Mau";
    const subtitle = "Hola soy un subtitulo";

    test('render component', () => {
        // ARRANGE
        const title = "Hola, Soy Mau";
        const {container} = render(<FirstApp title={title}/>);

        // ACT
        

        // ASSERT
        expect(container).toMatchSnapshot();
    });

    test('loads titles', () => {        
        // ARRANGE
        render(<FirstApp title={title} sub={subtitle}/>);
        
        // ACT
        const titleNode = screen.getByTestId('FirstApp.title');
        const subtitleNodes = screen.getAllByTestId('FirstApp.subtitle');

        // ASSERT
        expect(screen.getByText(title)).toBeTruthy();
        expect(titleNode).toHaveTextContent(title);
        expect(subtitleNodes.length).toBe(2);
    });
})