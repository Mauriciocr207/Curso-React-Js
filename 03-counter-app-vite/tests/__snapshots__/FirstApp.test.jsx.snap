// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirstApp test render component 1`] = `
<div>
  <h1
    data-testid="FirstApp.title"
  >
    Hola, Soy Mau
  </h1>
  <p
    data-testid="FirstApp.subtitle"
  >
    No hay subtitulo
  </p>
  <p
    data-testid="FirstApp.subtitle"
  >
    No hay subtitulo
  </p>
</div>
`;
